File: lib/LocalModules/DevOther/src/main/java/java/dev/other/retrofit/response/BaseResponse.java
package java.dev.other.retrofit.response;

/**
 * detail: 请求响应统一解析类
 * @author Ttt
 * <pre>
 *     主要用于统一解析后台返回 data、message、code 等
 * </pre>
 */
public class BaseResponse<T> {

    // 返回数据
    public           T         data;
    // 返回消息
    public           String    errorMsg; // message
    // 返回结果状态 ( 内部定义 )
    public           String    errorCode; // code
    // 请求结果
    public transient boolean   result;
    // 请求异常
    public transient Throwable exception;
}

File: lib/DevOther/src/main/java/ktx/dev/other/retrofit_rxjava/response/BaseResponse.kt
package ktx.dev.other.retrofit_rxjava.response

/**
 * detail: 请求响应统一解析类
 * @author Ttt
 * 主要用于统一解析后台返回 data、message、code 等
 */
class BaseResponse<T> {

    // 返回数据
    var data: T? = null

    // 返回消息
    var errorMsg: String? = null // message

    // 返回结果状态 ( 内部定义 )
    var errorCode: String? = null // code

    // 请求结果
    @Transient
    var result = false

    // 请求异常
    @Transient
    var exception: Throwable? = null
}